### PR TITLE
Update framework modules readme - Closes #3157

### LINKED ISSUE #3157: Update README for the framework modules
### Expected behavior
Describe the changes to the framework modules made in Lisk Core 1.7.0.

### Which version(s) does this affect? (Environment, OS, etc...)
1.6.0

### PR DESCRIPTION
**Pull request by Tschakki**
### What was the problem?

Changes in the Modules needed to be reflected int he README.

### How did I fix it?

Updated README file of Modules:
- Added configuration part.

### Review checklist

* The PR resolves #3157 
* All new code is covered with unit tests
* All new code was formatted with Prettier
* Linting passes
* Tests pass
* Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
* Documentation has been added/updated

### CODE REVIEW

File: framework/src/modules/README.md [hunk 1]
```diff
@@ -20,8 +21,8 @@ Core Modules are shipped along with the Lisk Core distribution itself. These mod
 
 #### List of Core Modules
 
-- **Chain Module:** handles all events and actions, that are related to the blockchain system.
-- **HTTP API Module:** provides API endpoints, that enable users and other programs to communicate with the Lisk blockchain through the API.
+* **Chain Module:** handles all events and actions, that are related to the blockchain system.
+* **HTTP API Module:** provides API endpoints, that enable users and other programs to communicate with the Lisk blockchain through the API.
```
Comment (nazarhussain) [inline framework/src/modules/README.md]: Please add the network module as well.

File: framework/src/modules/README.md [hunk 2]
```diff
@@ -120,6 +121,18 @@ export default class MyModule extends BaseModule {
 };
 ```
 
+## Module Configuration
+
+Configuration options for each module are located in `framework/src/modules/<module-name>/defaults/config.js`.
+
+Each `config.js` file consists of 2 parts:
+
+1. A general schema description of all available config options.
```
Comment (nazarhussain) [inline framework/src/modules/README.md]:
```suggestion
1. JSON-schema specification for all available config options
```

File: framework/src/modules/README.md [hunk 3]
```diff
@@ -120,6 +121,18 @@ export default class MyModule extends BaseModule {
...
+2. Default values for the available config options for this specific module.
+
+To customize the predefined defaults values, please don't change the default values in these files directly, as they will be overwritten on software updates.
```
Comment (nazarhussain) [inline framework/src/modules/README.md]:
```suggestion
Please don't change the default values in these files directly as they will be overwritten on software updates, instead define the custom configuration options inside your blockchain application.
```

File: framework/src/modules/README.md [hunk 4]
```diff
@@ -120,6 +121,18 @@ export default class MyModule extends BaseModule {
...
+Instead, define the custom configuration options inisde your blockchain application, and let it overwrite the default config options.
```
Comment (nazarhussain) [inline framework/src/modules/README.md]:
```suggestion
```

File: framework/src/modules/README.md [hunk 5]
```diff
@@ -146,11 +159,11 @@ A modules' life cycle consists of following events in the right order:
...
+* _module_:registeredToBus
+* _module_:loading:started
+* _module_:loading:finished
 
 **Unloading**
 
-- _module_:unloading:started
-- _module_:unloading:finished
+* _module_:unloading:started
+* _module_:unloading:finished
```
Comment (nazarhussain) [inline framework/src/modules/README.md]: @Tschakki The implementation for these events are not done yet. I had created issue for reference.

Comment (Tschakki) [reply framework/src/modules/README.md]: @nazarhussain Which events exactly?
Can you point me to the issue?

Comment (Tschakki) [reply framework/src/modules/README.md]: nvm I found it ! For reference: #3632

**Pull request merged**